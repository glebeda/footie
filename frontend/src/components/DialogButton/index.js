import React from 'react'
import Button from '@mui/material/Button'
import { styled } from '@mui/material/styles'

const StyledButton = styled(Button)(({ theme, variant }) => ({
  margin: theme.spacing(1),
  ...(variant === 'outlined' && {
    border: `1px solid ${theme.palette.primary.main}`
  }),
  ...(variant === 'contained' && {
    backgroundColor: theme.palette.primary.main,
    color: theme.palette.common.white,
    '&:hover': {
      backgroundColor: theme.palette.primary.dark
    }
  })
}))

const DialogButton = ({ children, variant, onClick, ...props }) => {
  return (
    <StyledButton variant={variant} onClick={onClick} {...props}>
      {children}
    </StyledButton>
  )
}

export default DialogButton
