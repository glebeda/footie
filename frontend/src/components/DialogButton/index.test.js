import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import DialogButton from './index';

describe('DialogButton', () => {
  const buttonText = "Click Me!";
  const onClickMock = jest.fn();

  it('renders correctly with variant="outlined"', () => {
    const { asFragment } = render(
      <DialogButton variant="outlined" onClick={onClickMock}>
        {buttonText}
      </DialogButton>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with variant="contained"', () => {
    const { asFragment } = render(
      <DialogButton variant="contained" onClick={onClickMock}>
        {buttonText}
      </DialogButton>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('calls onClick callback when clicked', () => {
    render(
      <DialogButton variant="outlined" onClick={onClickMock}>
        {buttonText}
      </DialogButton>
    );
    fireEvent.click(screen.getByText(buttonText));
    expect(onClickMock).toHaveBeenCalledTimes(1);
  });
});
