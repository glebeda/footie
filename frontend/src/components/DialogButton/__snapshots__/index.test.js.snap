// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogButton renders correctly with variant="contained" 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1ynnmgh-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Click Me!
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`DialogButton renders correctly with variant="outlined" 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-15rt0rr-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Click Me!
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
